imports/plugins/included/product-variant/client/containers/searchModalContainer.js

```diff
@@ -0,0 +1,226 @@
+import React, { Component } from "react";
+import { Meteor } from "meteor/meteor";
+import { Tracker } from "meteor/tracker";
+import _ from "lodash";
+import { Reaction } from "/client/api";
+import { composeWithTracker } from "/lib/api/compose";
+import * as Collections from "/lib/collections";
+import SearchModal from "../components/searchModal";
+
+class SearchModalContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collection: "products",
+      value: "",
+      tagResults: [],
+      productResults: [],
+      orderResults: [],
+      accountResults: []
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleAccountClick = this.handleAccountClick.bind(this);
+    this.handleOrderClick = this.handleOrderClick.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.dep = new Tracker.Dependency;
+  }
+
+  componentDidMount() {
+    Tracker.autorun(() => {
+      this.dep.depend();
+      const searchCollection = this.state.collection;
+      const searchQuery = this.state.value;
+      this.subscription = Meteor.subscribe("SearchResults", searchCollection, searchQuery, []);
```

Comment:
Because the subscription needs to get called every time state value changes (e.g. when you search for something/ change tabs from products to accounts etc).